### PR TITLE
enabling amc-mco on infra

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Signed-off-by: greg pereira <grpereir@redhat.com>

Addresses most of: https://github.com/operate-first/apps/issues/2337

/cc @HumairAK

### CODE REVIEW

File: cluster-scope/overlays/prod/moc/infra/kustomization.yaml
Comment (Gregory-Pereira) [inline cluster-scope/overlays/prod/moc/infra/kustomization.yaml:50]: this was just alphabetizing resources ... ignore

**Comment by Gregory-Pereira** [comment]
Follow on actions to this are creating the two secrets in vault once this pr enables the ESO for this namespace. Secrets in question:
1. multiclusterhub-operator-pull-secret 
2. thanos-object-storage

File: cluster-scope/overlays/prod/moc/infra/multiclusterobservability/kustomization.yaml
Comment (tumido) [inline cluster-scope/overlays/prod/moc/infra/multiclusterobservability/kustomization.yaml:5]: I don't think we expect multiple resources of the same kind in this overlay... Either please move it to the `cluster-scope/base` or remove the `infra/multiclusterobservability/kustomization.yaml` and include it directly as `infra/multiclusterobservability/observability.yaml` in `infra/kustomization.yaml`

File: cluster-scope/overlays/prod/moc/infra/externalsecrets/thanos-object-storage.yaml
Comment (tumido) [inline cluster-scope/overlays/prod/moc/infra/externalsecrets/thanos-object-storage.yaml:4]: It seems the NESE on Infra is a longer journey than expected.. let's point it to AWS for now..

https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.5/html-single/observability/index#enable-observability:~:text=For%20Amazon%20S3%20or%20S3%20compatible%2C%20your%20secret%20might%20resemble%20the%20following%20file%3A

https://github.com/operate-first/apps/pull/2286#pullrequestreview-1091209654

Comment (Gregory-Pereira) [reply cluster-scope/overlays/prod/moc/infra/externalsecrets/thanos-object-storage.yaml:4]: @tumido, noted, I have updated [my follow on action comment](https://github.com/operate-first/apps/pull/2338#issuecomment-1231934112). That being said this doesn't change any of the manifests in this pr, simply my steps after this gets merged. Can I confirm that you mean Balrog s3 when you say "lets point it to AWS for now..."?

**Comment by Gregory-Pereira** [comment]
Comments have been addressed, ready for re-review when you have time @tumido

File: cluster-scope/overlays/prod/moc/infra/externalsecrets/thanos-object-storage.yaml
Comment (tumido) [reply cluster-scope/overlays/prod/moc/infra/externalsecrets/thanos-object-storage.yaml:4]: I mean let's create a new AWS bucket + creds in the name of the operate first account and point it to there. I've just created one for you, please verify:

https://vault-ui-vault.apps.smaug.na.operate-first.cloud/ui/vault/secrets/k8s_secrets/show/moc/infra/open-cluster-management-observability/thanos-object-storage

**Review by tumido — APPROVED** [review]
/hold for @Gregory-Pereira

/lgtm
/approve

**Comment by tumido** [comment]
@Gregory-Pereira please rebase and unhold if you want this to get merged :slightly_smiling_face:

**Pull request merged**
